lsp/base_service_client/base_capabilities.go

```diff
@@ -0,0 +1,111 @@
+package base
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/konveyor/analyzer-lsp/lsp/protocol"
+	"github.com/konveyor/analyzer-lsp/provider"
+	"go.lsp.dev/uri"
+	"gopkg.in/yaml.v2"
+)
+
+// Type aliases to make the function definitions shorter
+
+type ctx = context.Context
+type resp = provider.ProviderEvaluateResponse
+type base = HasLSPServiceClientBase
+
+// Technically not necessary
+type NoOpCondition struct{}
+
+// A simple no-op function. Returns a blank response
+func EvaluateNoOp[T base](t T, ctx ctx, cap string, info []byte) (resp, error) {
+	return resp{}, nil
+}
+
+// Generic referenced condition
+type ReferencedCondition struct {
+	Referenced struct {
+		Pattern string `yaml:"pattern"`
+	} `yaml:"referenced"`
+}
+
+// EvaluateReferenced evaluates references to a given entity based on a query
+// pattern. The function uses the provided query pattern to find references to
+// the specified entity within the workspace, filters out references in certain
+// directories, and returns a list of incident contexts associated with these
+// references.
+func EvaluateReferenced[T base](t T, ctx ctx, cap string, info []byte) (resp, error) {
+	sc := t.GetLSPServiceClientBase()
+
+	var cond ReferencedCondition
+	err := yaml.Unmarshal(info, &cond)
+	if err != nil {
+		return resp{}, fmt.Errorf("error unmarshaling query info")
+	}
+
+	query := cond.Referenced.Pattern
+	if query == "" {
+		return resp{}, fmt.Errorf("unable to get query info")
+	}
+
+	symbols := sc.GetAllDeclarations(ctx, sc.BaseConfig.WorkspaceFolders, query)
+
+	incidents := []provider.IncidentContext{}
+	incidentsMap := make(map[string]provider.IncidentContext) // Remove duplicates
+
+	for _, s := range symbols {
+		references := sc.GetAllReferences(ctx, s.Location.Value.(protocol.Location))
+
+	ITERATE_REFERENCES:
```

Comment:
remove the goto please :)

Reply:
Give it a chance! Alas, the goto is merely misunderstood...

Removed it